backend/src/test/java/codezap/like/repository/LikesJpaRepositoryTest.java

```diff
@@ -0,0 +1,153 @@
+package codezap.like.repository;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import codezap.category.repository.CategoryRepository;
+import codezap.fixture.CategoryFixture;
+import codezap.fixture.MemberFixture;
+import codezap.fixture.TemplateFixture;
+import codezap.global.repository.JpaRepositoryTest;
+import codezap.like.domain.Likes;
+import codezap.member.domain.Member;
+import codezap.member.repository.MemberRepository;
+import codezap.template.domain.Template;
+import codezap.template.repository.TemplateRepository;
+
+@JpaRepositoryTest
+class LikesJpaRepositoryTest {
+
+    @Autowired
+    private LikesRepository likesRepository;
+
+    @Autowired
+    private MemberRepository memberRepository;
+
+    @Autowired
+    private CategoryRepository categoryRepository;
+
+    @Autowired
+    private TemplateRepository templateRepository;
+
+    @Nested
+    @DisplayName("좋아요 저장 테스트")
+    class SaveTest {
+
+        @Test
+        @DisplayName("성공")
+        void success() {
+            Member member = memberRepository.save(MemberFixture.getFirstMember());
+            Template template = templateRepository.save(TemplateFixture.get(
+                    member,
+                    categoryRepository.save(CategoryFixture.getFirstCategory())
+            ));
+
+            likesRepository.save(new Likes(null, template, member));
+
+            assertThat(likesRepository.countByTemplate(template)).isEqualTo(1);
```

Comment:
save에 대한 repository 테스트인데 `countByTemplate`이 사용되는 것이 어색한 것 같아요.
```suggestion
            Likes likes = likesRepository.save(new Likes(null, template, member));

            assertAll(
                () -> assertThat(likes.getMember).isEqualTo(member),
                () -> assertThat(likes.getTemplate).isEqualTo(template)
```
추가로 저희 직접 작성한 jpa 쿼리 메소드에 대해서만 테스트를 진행하기로 하지 않았었나요??

Reply:
> save에 대한 repository 테스트인데 countByTemplate이 사용되는 것이 어색한 것 같아요.

이 부분은 개인적으로는 저장, 삭제, 조회 등을 테스트할때 어쩔 수 없이 발생하는 모순이라 생각해요.
결국 제가 확인하고 싶었던 것은 `save()` 를 저장했을 때 반환값이 아닌, 올바르게 값이 들어있다는 것을 확인하고 싶었습니다.

> 추가로 저희 직접 작성한 jpa 쿼리 메소드에 대해서만 테스트를 진행하기로 하지 않았었나요??

그렇긴 한데, 빠른 시간 내에 기능을 개발하려다 보니 마음이 불안해서...
테스트를 작성해 버리고 말았어요...ㅎㅎ

Reply:
테스트가 많은 건 좋죠ㅎㅎㅎ 좋습니다!!